packages/terminal/src/browser/base/terminal-widget.ts

```diff
@@ -30,13 +30,79 @@ export interface TerminalExitStatus {
     readonly code: number | undefined;
 }
 
+export type TerminalLocationOptions = TerminalLocation | TerminalEditorLocation | TerminalSplitLocation;
+
+export enum TerminalLocation {
+    Panel = 1,
+    Editor = 2
+}
+
+export interface TerminalEditorLocation {
+    readonly viewColumn: number;
+    readonly preserveFocus?: boolean;
+}
+
+export interface TerminalSplitLocation {
+    readonly parentTerminal: string;
+}
+
+export enum ViewColumn {
```

Comment:
@xai I don't believe that `ViewColumn` should be defined in `@theia/terminal` but rather `@theia/core` since it is generic enough to apply to the `ApplicationShell` and the main-area.

 At the very least it should be defined in `@theia/editor` if you disagree.

Reply:
You are right, `ViewColumn` is definitely generic and might be used by other components in the future! I will move it.

Reply:
Moved to `core` in https://github.com/eclipse-theia/theia/pull/12006/commits/52abd5e2903e4ce46eac7b8c570e310c53eef776.